pkg/kubeadm/master/manifests.go

```diff
@@ -0,0 +1,195 @@
+		}),
+		// TODO bind-mount certs in
+		"kube-apiserver": componentPod(api.Container{
+			Name:  "kube-apiserver",
+			Image: params.EnvParams["hyperkube_image"],
+			Command: []string{
+				"/hyperkube",
+				"apiserver",
+				"--address=127.0.0.1",
+				"--etcd-servers=http://127.0.0.1:2379",
+				"--cloud-provider=fake", // TODO parametrise
+				"--admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota",
+				SERVICE_CLUSTER_IP_RANGE,
+				"--service-account-key-file=/etc/kubernetes/pki/apiserver-key.pem",
+				"--client-ca-file=/etc/kubernetes/pki/ca.pem",
+				"--tls-cert-file=/etc/kubernetes/pki/apiserver.pem",
+				"--tls-private-key-file=/etc/kubernetes/pki/apiserver-key.pem",
+				"--secure-port=443",
+				"--allow-privileged",
+				COMPONENT_LOGLEVEL,
+				"--token-auth-file=/etc/kubernetes/pki/tokens.csv",
+			},
+			VolumeMounts:  []api.VolumeMount{pkiVolumeMount()},
+			LivenessProbe: componentProbe(8080, "/healthz"),
+			Resources:     componentResources("250m"),
+		}, pkiVolume(params)),
+		"kube-controller-manager": componentPod(api.Container{
+			Name:  "kube-controller-manager",
+			Image: params.EnvParams["hyperkube_image"],
+			Command: []string{
+				"/hyperkube",
+				"controller-manager",
+				"--leader-elect",
+				MASTER,
+				CLUSTER_NAME,
+				"--root-ca-file=/etc/kubernetes/pki/ca.pem",
+				"--service-account-private-key-file=/etc/kubernetes/pki/apiserver-key.pem",
+				"--cluster-signing-cert-file=/etc/kubernetes/pki/ca.pem",
+				"--cluster-signing-key-file=/etc/kubernetes/pki/ca-key.pem",
+				"--insecure-experimental-approve-all-kubelet-csrs-for-group=system:kubelet-bootstrap",
+				COMPONENT_LOGLEVEL,
+			},
+			VolumeMounts:  []api.VolumeMount{pkiVolumeMount()},
+			LivenessProbe: componentProbe(10252, "/healthz"),
+			Resources:     componentResources("200m"),
+		}, pkiVolume(params)),
+		"kube-scheduler": componentPod(api.Container{
+			Name:  "kube-scheduler",
+			Image: params.EnvParams["hyperkube_image"],
+			Command: []string{
+				"/hyperkube",
+				"scheduler",
+				"--leader-elect",
+				MASTER,
+				COMPONENT_LOGLEVEL,
+			},
+			LivenessProbe: componentProbe(10251, "/healthz"),
+			Resources:     componentResources("100m"),
+		}),
+	}
+
+	manifestsPath := path.Join(params.EnvParams["prefix"], "manifests")
+	if err := os.MkdirAll(manifestsPath, 0700); err != nil {
+		return err
+	}
+	for name, spec := range staticPodSpecs {
+		serialized, err := json.MarshalIndent(spec, "", "  ")
+		if err != nil {
+			return err
+		}
+		if err := cmdutil.DumpReaderToFile(bytes.NewReader(serialized), path.Join(manifestsPath, name+".json")); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func pkiVolume(params *kubeadmapi.BootstrapParams) api.Volume {
+	return api.Volume{
+		Name: "pki",
+		VolumeSource: api.VolumeSource{
+			HostPath: &api.HostPathVolumeSource{Path: params.EnvParams["host_pki_path"]},
```

Comment:
Future note: as we move to HA, this'll get more complicated as we'll have to find a way to HA any CA functionality.